### PR TITLE
kubernetes: 1.14.3 -> 1.15.3 (with revert of module systemd dependencies)

### PR DESCRIPTION
**Pull request by johanot**
###### Motivation for this change
1. To get kubernetes bumped to v1.15 before 19.09 freeze.
2. To make the NixOS module work, be compatible with 1.15 and make current rbac and dns tests pass.

#### Note: This PR reverts changes introduced in #56789.
While I think it is a good idea to use systemd to sync startup of components, it has unfortunately introduced some problems in practice (e.g. #60687, #61135).
I actually have a branch here, on which I tried making the upgrade and tests work with the current module: https://github.com/johanot/nixpkgs/commits/kubernetes-1.15.2 .. Unfortunately, there are still race condition issues causing systemd restart-limits to be breached and Flannel vxlan connections be broken, among other things.

After spending a lot of time thinking about and working on the current implementation, considering the trade-offs, I ultimately think it's better to let eventual consistency rule and let components restart until the cluster is fully healthy. While it slows down startup time and causes more restarts, it reduces code complexity and untangles the systemd dependency web - the latter being invaluable in debugging situations.

@calbrecht I'm sorry I didn't state this argument explicitly before now. Are you ok with a revert at this time? 

#### 19.09 vs. 20.03
I know this PR might conflict with #66496. I just want to make sure that we get _something_ working in 19.09. And then, when we hit October or something like that, let's discuss the _real_ future for kubernetes in nixpkgs.

cc @offlinehacker @srhb

###### Things done

RBAC and DNS tests run clean locally.

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by srhb** [comment]
I am in favor of this. I was also surprised with the difficulties in upgrading because of the systemd dependencies, even though I used to think they were a good idea. I guess we learned something.

And yes, a radical redesign of the k8s infrastructure before freeze is not really viable, but I agree with an extremely simplified module being the way forward afterwards.

**Comment by srhb** [comment]
@globin In lieu of @calbrecht's feedback, how do you feel about the module dependency reverts? I tried to catch you on -dev the other day to discuss it, but I have the feeling you're pretty busy :grin:

If no one voices any serious concerns about this, I'm inclined to merge.

**Review by globin — APPROVED** [review]

**Comment by srhb** [comment]
Rebased to fix the pname conflict.

**Comment by arianvp** [comment]
Yep I'm also in favor of doing this.  It will make startups more noisy for sure but at least it will work reliably. We can then slowly contemplate maybe introducing some of these ordering constraints again in some shape or another.

File: nixos/modules/services/cluster/kubernetes/kubelet.nix [hunk 1]
```diff
@@ -308,56 +294,6 @@ in
           '';
           WorkingDirectory = top.dataDir;
         };
-        unitConfig.ConditionPathExists = kubeletPaths;
-      };
-
-      systemd.paths.kubelet = {
```
Comment (arianvp) [inline nixos/modules/services/cluster/kubernetes/kubelet.nix]: Note that this didn't do what we probably wanted in the first place.  

This `systemctl start kubelet.service` any time either `PathExists=` or `PathChanged=` is matched.  Think of `.path` unit like `.timer` units.  A `.timer` unit isn't active for the lifetime for when the Calendar clause is active, it is _always_ active and gets pulled in by `timers.target` It merely activates another unit when the `Calendar` clause is matched.  Similarily,  a `.path` unit is part of `paths.target` and is _always active_ and then Starts a `.service` whenever some conditions changes.  It's a mistake I have made a lot too when first looking at `.path` units. 

Another example is a `.socket` unit. a `.socket` unit is always active (gets pulled in by `sockets.target`) but starts a `.service` whenever activity on a socket occurs.

Comment (srhb) [reply nixos/modules/services/cluster/kubernetes/kubelet.nix]: I hadn't even noticed that. Definitely something to keep in mind. That said, I'm not sure we *can* come up with an ordering scheme that's reasonably simple and still works in disparate environments, with all the combinations of daemons we can think of. We'll see. :)

**Pull request merged**